permissions/perUserMfaState/grantPermission.ps1

```diff
@@ -0,0 +1,322 @@
+                Uri         = "$($baseUri)/beta/users/$($actionContext.References.Account)/authentication/requirements"
+                Method      = "PATCH"
+                Body        = ($updatePerUserMfaStateBody | ConvertTo-Json -Depth 10)
+                Verbose     = $false
+                ErrorAction = "Stop"
+            }
+
+            Write-Verbose "SplatParams: $($updatePerUserMfaStateSplatParams | ConvertTo-Json)"
+
+            if (-Not($actionContext.DryRun -eq $true)) {
+                # Add Headers after printing splat
+                $updatePerUserMfaStateSplatParams['Headers'] = $headers
+
+                $updatedPerUserMfaState = Invoke-RestMethod @updatePerUserMfaStateSplatParams
+
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "$state perUserMfaState [$($actionContext.References.Permission.Name)] for account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Old value: [$($currentPerUserMfaState)]. New value: [$($state)]."
+                        IsError = $false
+                    })
+            }
+            else {
+                Write-Warning "DryRun: Would set perUserMfaState to [$($state)] for account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Old value: [$($currentPerUserMfaState)]. New value: [$($state)]."
+            }
+            #endregion Update PerUserMfaState
+
+            break
+        }
+
+        "NoChanges" {
+            #region No changes
+            $actionMessage = "skipping setting perUserMfaState to [$($state)] for account"
+
+            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                    # Action  = "" # Optional
+                    Message = "Skipped setting perUserMfaState to [$($state)] for account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Old value: [$($currentPerUserMfaState)]. New value: [$($state)]. Reason: No changes."
+                    IsError = $false
+                })
+            #endregion No changes
+
+            break
+        }
+    }
+    #endregion Process
+}
+catch {
+    $ex = $PSItem
+    if ($($ex.Exception.GetType().FullName -eq 'Microsoft.PowerShell.Commands.HttpResponseException') -or
+        $($ex.Exception.GetType().FullName -eq 'System.Net.WebException')) {
+        $errorObj = Resolve-MicrosoftGraphAPIError -ErrorObject $ex
+        $auditMessage = "Error $($actionMessage). Error: $($errorObj.FriendlyMessage)"
+        $warningMessage = "Error at Line [$($errorObj.ScriptLineNumber)]: $($errorObj.Line). Error: $($errorObj.ErrorDetails)"
+    }
+    else {
+        $auditMessage = "Error $($actionMessage). Error: $($ex.Exception.Message)"
+        $warningMessage = "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($ex.Exception.Message)"
+    }
+
+    Write-Warning $warningMessage
```

Comment:
Waarschijnlijk even uitbreiden met 'Unable to set $state for person X. Error message: '

Reply:
Lijkt me overbodig, dit blijkt al uit de actie